milestone-2.md

```diff
@@ -0,0 +1,380 @@
+# AST Structure
+
+### enum ScalarType:
+
+    Void
+    Int
+    Char
+    Bool
+    Str
+    Identifier(String)
+
+### enum Type
+
+    Base(ScalarType)
+    Array(Box<Type>)
+
+### struct CompilationUnit
+
+    classes: Vec<Class>
+    main: Vec<Stmt>
+
+### struct Class
+
+    id: String
+    members: Vec<ClassMember>
+
+### enum ClassMember
+
+    DataMember {
+        is_static: bool
+        is_private: bool
+        stmt: Stmt
+    }
+
+    MethodMember {
+        is_static: bool
+        is_private: bool
+        return_type: Type
+        id: String
+        params: Vec<Parameter>
+        body: Vec<Stmt>
+    }
+
+    Constructor {
+        id: String
+        params: Vec<Parameter>
+        body: Vec<Stmt>
+    }
```

Comment:
This is fine. You might consider if you want to make constructors methods or keep them as separate data constructors.